docs/developer/transform.rst

```diff
@@ -70,11 +75,34 @@ Note that the template argument ``std::tuple<bool>`` is equivalent to ``std::tup
 Recommended usage
 -----------------
 
-For improved testability and maintainability it is recommended to define the operator stand-alone instead of inline.
-The list of supported types can also be provided in this manner.
+- For improved testability and maintainability it is recommended to define the operator stand-alone instead of inline.
+- Specifying types on the element wise operator allows them to be omitted from 
+``transform`` and ``transform_in_place``.
+- Use ``arg_list`` to define list of supported type combinations as shown below.
+
+.. code-block:: cpp
+
+  // This will accept a pair of (double, bool) / (float, int) / ...etc.
+  overloaded{arg_list<std::tuple<double, bool>,
+                      std::tuple<float, int>,
+                      std::tuple<long, double>  // Any other type pairs you support
+                      >
+                      [](auto &a, auto &b) { // Get double/bool...etc. pairs here},
+                      [](const units::Unit &a, const units::Unit &b) { // Unit handling }
+             };
 
-- Use ``arg_list`` to define list of supported type combinations.
 - Flags in ``transform_flags.h`` can be used to, e.g., prevent code generation if an argument has variances.
+- If pairs of the same type are required use ``pair_self_t`` for example:
+
+.. code-block:: cpp
+
+  // Equivalent to tuple<bool, bool>, tuple<double, double> ....
+  transform<pair_self_t<bool, double, int, ...etc>>(...);
+
+- If a new value is initialized or different types could be passed use ``decltype`` with static casts to coerce
+the compiler to use the correct type (rather than implicit deduction). This prevents dynamic casts which could
+fail at runtime (static cast will fail at compilation), and emit warnings on MSVC.
```

Comment:
Not sure what you are talking about here? Where is a dynamic cast happening? `transform` will create the output variable based on the return type of the functor, that's it. If you need precise control over the created output, you may want to use `static_cast`, but I feel that is adding noise to the documentation. Maybe just be more explicit about how the return type dictates the element type of the created variable?

Reply:
This was from me being mislead that we had implicit conversions (which was actually bitwise comparisons) on a separate build whilst writing these docs

I'm going to drop this point all together now